### ISSUE TITLE
Agree on concrete CoLiS syntax

### ISSUE THREAD

**Issue by benozol** [colis-anr/colis-language#4]
A draft based on the old parser is in `src/colisParser.mly`.

The string syntax could be simplified.

**Comment by Niols** [comment]
Yes, there are a lot of things that could and should be simplified imho, but this will be a long (opinionated) discussion

**Comment by benozol** [comment]
Oh that may be very true. We could just assign the issue to a syntactic dictator. BTW the string current syntax is @claudemarche ’s proposal.

**Comment by Niols** [comment]
It is, but it was also in the context where there were strings and lists (as values) and the expressions were all much more complicated. With the new type, there is probably a way to simplify all that a lot.

**Comment by benozol** [comment]
Here is an example from Claude’s proposal
```
begin
  ab := 'a b';
  c := ' c';
  d := 'd';
  for s in [{ab c d ' e'}; {'f g'}; 'h i'] do
    echo [s]
  done
end
```

Braces mark 'DontSplit' and the output is

```
a b cd e
f g
h
i

```

**Comment by Niols** [comment]
Isn't it a bit weird that we have to add something when we want to avoid splitting? I know this is how it goes in Shell, but I'm not sure it's a good example to follow ;-)

**Comment by claudemarche** [comment]
Usually I find these tons of e-mails very hard to follow. Though, for this one I can add my two cents: the syntax with curly braces {..} certainly evokes something that is more atomic, so naturally non-splitting. If we want a concrete syntax where  non-splitting is the default, then the way to explicitly require splitting should be more like using a keyword 'split'

- Claude
